File: android/app/src/main/java/io/flutter/app/FlutterApplication.java
package io.flutter.app;

import io.flutter.embedding.android.FlutterActivity;
import io.flutter.embedding.engine.FlutterEngine;
import io.flutter.plugins.GeneratedPluginRegistrant;
import android.app.Application;

/**
 * Flutter's default Application class for v2 embedding backward compatibility.
 * 
 * This can be removed once all plugins requiring the embedding v1 are migrated
 * to the embedding v2.
 */
public class FlutterApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
    }
} 

File: android/app/src/main/kotlin/com/example/smch_mobile/FlutterApplication.kt
package com.example.smch_mobile

import io.flutter.embedding.engine.FlutterEngine
import io.flutter.embedding.engine.dart.DartExecutor.DartEntrypoint

/**
 * Application class that properly implements Flutter's v2 embedding.
 * This is referenced by the AndroidManifest.xml via ${applicationName}
 */
class FlutterApplication : android.app.Application() {
    override fun onCreate() {
        super.onCreate()
        // Flutter v2 embedding handles plugin registration automatically
    }
}